pkg/secrets/repositoryserver/azure_secrets.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repositoryserver
+
+import (
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ RepositoryServerSecret = &Azure{}
+
+type Azure struct {
+	storageLocation *corev1.Secret
+}
+
+func NewAzureLocation(secret *corev1.Secret) *Azure {
+	return &Azure{
+		storageLocation: secret,
+	}
+}
+
+func (l *Azure) Validate() error {
+	if _, ok := l.storageLocation.Data[BucketKey]; !ok {
```

Comment:
I think the fields here need to be changes.
For azure, it should be:
- container name
and rest should be under credentials secret.

Reply:
You can refer to https://github.com/kanisterio/kanister/blob/master/pkg/kopia/command/storage/azure_args.go